### ISSUE TITLE
Project with Firebase and gRPC-Core third library

### ISSUE THREAD

**Issue by ernesto** [firebase/firebase-ios-sdk#93]
### [READ] Step 1: Are you in the right place?

 YES

### [REQUIRED] Step 2: Describe your environment

  * Xcode version: 8.3.2
  * Firebase SDK version: 4.0.0
  * Library version: 4.0.0
  * Firebase Product: core, analytics, message, auth, database, remote config

### [REQUIRED] Step 3: Describe the problem

Starting with firebase 3.13.X, application give a BAD_ACCESS on gRPC-Core/pb_decode.c on line 796, just later [FIRApp configure] on App Delegate. Using firebase 3.12 work fine. That's a Protobuffer issue?

#### Steps to reproduce:

Just put firebase version 3.13.0 or greater with pod 'gRPC-ProtoRPC' latest version on same project and the issue occurs

**Comment by paulb777** [comment]
@ernesto I added pod 'gRPC-ProtoRPC' to the [Analytics quickstart](https://github.com/firebase/quickstart-ios/tree/master/analytics) and it runs fine in the simulator with Xcode 8.3.2 and Firebase 4.0.2.

Please provide more details about how to reproduce.

**Comment by ernesto** [comment]
@paulb777 Hi. In fact, in the Analytics quickstart with  pod 'gRPC-ProtoRPC' don't have issues on first install.
But i notice that's just occurs when is removed use_frameworks! in podfile and install with a previous working example installed on simulator or device.
If i run with a clean installation, i not get a crash, but analytics won't work.

```
2017-06-23 18:11:53.099 AnalyticsExample[68099] <Debug> [Firebase/Analytics][I-ACS023073] Debug mode is enabled. Marking event as debug and real-time. Event name, parameters: screen_view (_vs), {
	    firebase_event_origin (_o) = auto;
	    firebase_realtime (_r) = 1;
	    firebase_previous_screen (_pn) = A;
	    firebase_screen_class (_sc) = FoodPickerViewController;
	    firebase_screen_id (_si) = -969582124148602935;
	    firebase_previous_class (_pc) = ViewController;
	    firebase_debug (_dbg) = 1;
	    firebase_previous_id (_pi) = -969582124148602936;
	}
2017-06-23 18:11:53.102 AnalyticsExample[68099] <Warning> [Firebase/Analytics][I-ACS012011] Received empty measurement bundle metadata for getting dictionary representation
2017-06-23 18:11:53.104 AnalyticsExample[68099] <Error> [Firebase/Analytics][I-ACS023068] Data loss. Cannot record event metadata. Error: Error Domain=com.google.gmp.measurement.ErrorDomain Code=2 "Failed to insert/update unknown data type" UserInfo={NSLocalizedDescription=Failed to insert/update unknown data type}
2017-06-23 18:11:53.105 AnalyticsExample[68099] <Error> [Firebase/Analytics][I-ACS034010] Transaction failed
2017-06-23 18:11:53.106 AnalyticsExample[68099] <Debug> [Firebase/Analytics][I-ACS023024] No data to upload. Upload task will not be scheduled
2017-06-23 18:11:53.107 AnalyticsExample[68099] <Debug> [Firebase/Analytics][I-ACS002001] Search Ad campaign report alarm fired
2017-06-23 18:11:53.107 AnalyticsExample[68099] <Debug> [Firebase/Analytics][I-ACS002003] Search Ad campaign report alarm canceled
2017-06-23 18:11:53.114 AnalyticsExample[68099] <Debug> [Firebase/Analytics][I-ACS032006] Scheduling Search Ad Report timer
2017-06-23 18:11:53.114 AnalyticsExample[68099] <Debug> [Firebase/Analytics][I-ACS002002] Search Ad campaign report alarm scheduled to fire in approx. (s): 86399.99112701416
2017-06-23 18:11:54.589 AnalyticsExample[68099] <Error> [Firebase/Analytics][I-ACS032010] Cannot send request for Search Ad campaign attribution. Error: Error Domain=ADClientErrorDomain Code=0 "Unable to retrieve attribution details" UserInfo={NSLocalizedDescription=Unable to retrieve attribution details}
2017-06-23 18:11:54.591 AnalyticsExample[68099] <Warning> [Firebase/Analytics][I-ACS032005] Search Ad Campaign attribution alarm is already scheduled
2017-06-23 18:12:03.061 AnalyticsExample[68099] <Debug> [Firebase/Analytics][I-ACS002001] Session timer fired
2017-06-23 18:12:03.062 AnalyticsExample[68099] <Debug> [Firebase/Analytics][I-ACS002003] Session timer canceled
2017-06-23 18:12:03.063 AnalyticsExample[68099] <Debug> [Firebase/Analytics][I-ACS023051] Logging event: origin, name, params: auto, session_start (_s), {
	    firebase_event_origin (_o) = auto;
	    firebase_screen_class (_sc) = FoodPickerViewController;
	    firebase_screen_id (_si) = -969582124148602935;
	}
2017-06-23 18:12:03.064 AnalyticsExample[68099] <Debug> [Firebase/Analytics][I-ACS023073] Debug mode is enabled. Marking event as debug and real-time. Event name, parameters: session_start (_s), {
	    firebase_screen_id (_si) = -969582124148602935;
	    firebase_event_origin (_o) = auto;
	    firebase_screen_class (_sc) = FoodPickerViewController;
	    firebase_realtime (_r) = 1;
	    firebase_debug (_dbg) = 1;
	}
2017-06-23 18:12:03.068 AnalyticsExample[68099] <Warning> [Firebase/Analytics][I-ACS012011] Received empty measurement bundle metadata for getting dictionary representation
2017-06-23 18:12:03.070 AnalyticsExample[68099] <Error> [Firebase/Analytics][I-ACS023068] Data loss. Cannot record event metadata. Error: Error Domain=com.google.gmp.measurement.ErrorDomain Code=2 "Failed to insert/update unknown data type" UserInfo={NSLocalizedDescription=Failed to insert/update unknown data type}
2017-06-23 18:12:03.072 AnalyticsExample[68099] <Error> [Firebase/Analytics][I-ACS034010] Transaction failed
2017-06-23 18:12:03.072 AnalyticsExample[68099] <Debug> [Firebase/Analytics][I-ACS023024] No data to upload. Upload task will not be scheduled
```

**Comment by ernesto** [comment]
Is a Objective-C project.

**Comment by paulb777** [comment]
@ernesto Thanks for the report and follow up clarification!

I'm able to reproduce and will investigate with the team.

**Comment by paulb777** [comment]
We're targeting a fix for the next release after Firebase 4.0.3

**Comment by paulb777** [comment]
This issue is now fixed in Firebase 4.0.4 and gRPC 1.4.2.
